skyportal/models.py

```diff
@@ -300,9 +316,11 @@ def altitude(self, telescope, time):
 
 
 class Filter(Base):
-    query_string = sa.Column(sa.String, nullable=False, unique=False)
+    name = sa.Column(sa.String, nullable=False, unique=False)
+    stream_id = sa.Column(sa.ForeignKey("streams.id"), nullable=False)
```

Comment:
Shouldn't need this, as it is now specified by `group`, right? I.e., filters no longer store stream credentials.

Reply:
A group may have access to multiple streams and you define a filter in the group on a stream, so I think it should be there.

Reply:
In that case I think it makes more sense to enforce stream access on the filter level than the group level. Otherwise, you need to manually update each group when you remove a stream with, e.g., PID=4 (Gattini).

Reply:
I don't see why you would delete a stream. But even if you do, the update will happen automatically, see `test_post_new_filter_delete_group_deletes_filter` and `test_post_new_filter_delete_stream_deletes_filter` in `test_groups.py`.